terraform/eks/daemon/gpu/main.tf

```
@@ -706,12 +707,179 @@ resource "kubernetes_cluster_role_binding" "rolebinding" {
     namespace = "amazon-cloudwatch"
   }
 }
+# NVIDIA Device Plugin DaemonSet
+resource "kubernetes_daemonset" "nvidia_device_plugin" {
+  depends_on = [
+    kubernetes_namespace.namespace,
+    aws_eks_node_group.this,
+  ]
+  metadata {
+    name      = "nvidia-device-plugin-daemonset"
+    namespace = "kube-system"
+  }
+  spec {
+    selector {
+      match_labels = {
+        name = "nvidia-device-plugin-ds"
+      }
+    }
+    template {
+      metadata {
+        labels = {
+          name = "nvidia-device-plugin-ds"
+        }
+      }
+      spec {
+        toleration {
+          key      = "nvidia.com/gpu"
+          operator = "Exists"
+          effect   = "NoSchedule"
+        }
+        priority_class_name = "system-node-critical"
+        container {
+          name  = "nvidia-device-plugin-ctr"
+          image = "nvcr.io/nvidia/k8s-device-plugin:v0.17.0"
+          args  = ["--fail-on-init-error=false"]
+          security_context {
+            allow_privilege_escalation = false
+            capabilities {
+              drop = ["ALL"]
+            }
+          }
+          volume_mount {
+            name       = "device-plugin"
+            mount_path = "/var/lib/kubelet/device-plugins"
+          }
+        }
+        volume {
+          name = "device-plugin"
+          host_path {
+            path = "/var/lib/kubelet/device-plugins"
+          }
+        }
+      }
+    }
+  }
+}
+
+# Set up kubectl configuration for EKS cluster
+resource "null_resource" "kubectl" {
+  depends_on = [
+    aws_eks_cluster.this,
+    aws_eks_node_group.this,
+  ]
+  provisioner "local-exec" {
+    command = <<-EOT
+      ${local.aws_eks} update-kubeconfig --name ${aws_eks_cluster.this.name}
+      ${local.aws_eks} list-clusters --output text
+      ${local.aws_eks} describe-cluster --name ${aws_eks_cluster.this.name} --output text
+    EOT
+  }
+}
+
+# Wait for NVIDIA device plugin to be truly ready
+resource "null_resource" "wait_for_nvidia_ready" {
+  depends_on = [
+    null_resource.kubectl,
+    kubernetes_daemonset.nvidia_device_plugin
+  ]
+
+  provisioner "local-exec" {
+    command = <<-EOT
+      # Wait for NVIDIA device plugin to be ready
+      kubectl wait --for=condition=ready pod -l name=nvidia-device-plugin-ds -n kube-system --timeout=300s
+      
+      # Verify GPU resources are available on nodes
+      kubectl get nodes -o jsonpath='{.items[*].status.allocatable.nvidia\.com/gpu}' | grep -q "1"
+    EOT
+  }
+}
```

Can we add some more debugging info in case it fails to come up? kiro suggested:
```suggestion
resource "null_resource" "wait_for_nvidia_ready" {
  depends_on = [
    null_resource.kubectl,
    kubernetes_daemonset.nvidia_device_plugin
  ]

  provisioner "local-exec" {
    command = <<-EOT
      if ! kubectl wait --for=condition=ready pod -l name=nvidia-device-plugin-ds -n kube-system --timeout=300s; then
        echo "ERROR: NVIDIA device plugin failed to become ready"
        kubectl get pods -n kube-system -l name=nvidia-device-plugin-ds
        kubectl describe pods -n kube-system -l name=nvidia-device-plugin-ds
        exit 1
      fi

      if ! kubectl get nodes -o jsonpath='{.items[*].status.allocatable.nvidia\.com/gpu}' | grep -q "1"; then
        echo "ERROR: No GPU resources found on nodes"
        kubectl get nodes -o json | jq '.items[].status.allocatable'
        exit 1
      fi
    EOT
  }
}
```

---

Good idea!